hail/python/hailtop/batch/batch.py

```diff
@@ -262,9 +262,7 @@ async def _serialize_python_to_input_file(
             await self._fs.makedirs(os.path.dirname(code_path), exist_ok=True)
             await self._fs.write(code_path, pipe.getvalue())
 
-        code_input_file = self.read_input(code_path)
-
-        return code_input_file
+        return await self._async_read_input(code_path)
```

Comment:
This is the crucial part of the path here. Since `self.read_input` calls `validate_file`, it is a synchronous call that blocks on an event loop task. But it's inside the async `_serialize_python_to_input_file`, so we should have async all the way down.